### PR TITLE
[XHarness] Fix some issues in the BCL test generator.

### PR DESCRIPTION
**Pull request by mandel-macaque**
We had issues in the code that adds a type found in an assembly to
ensure that it was not removed by the linker. This resulted in some
assemblies having 0 tests.

Added the needed ignore for the corlib tests and system ones.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Adding do not merge since we know there is a AOT error when ran on devices, waiting on a fix from mono.

**Comment by akoeplinger** [comment]
The AOT error will be fixed by https://github.com/mono/mono/pull/13997

File: tests/xharness/AppRunner.cs [hunk 1]
```diff
@@ -448,14 +448,22 @@ public async Task<bool> TestsSucceeded (Log listener_log, bool timed_out, bool c
...
 			}
+
 			// parsing the human readable results
-			if (log.Contains ("Tests run")) {
+			if (resultLine != null) {
 				var tests_run = string.Empty;
-				var log_lines = log.Split ('\n');
 				var failed = false;
 				foreach (var line in log_lines) {
 					if (line.Contains ("Tests run:")) {
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply tests/xharness/AppRunner.cs]: I though about it, hated the O 2n.. but I wondered about the if statement.

**Comment by mandel-macaque** [comment]
Superseeded by https://github.com/xamarin/xamarin-macios/pull/5938

**Pull request closed without merging**